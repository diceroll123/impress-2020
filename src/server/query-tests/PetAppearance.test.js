const gql = require("graphql-tag");
const { query, getDbCalls } = require("./setup.js");

describe("PetAppearance", () => {
  it("loads for species and color", async () => {
    const res = await query({
      query: gql`
        query {
          petAppearance(speciesId: "54", colorId: "75", pose: HAPPY_FEM) {
            id

            species {
              id
              name
            }

            color {
              id
              name
              isStandard
            }

            layers {
              id
              imageUrl(size: SIZE_600)
              svgUrl
              zone {
                depth
              }
              restrictedZones {
                id
              }
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT * FROM pet_states
             WHERE pet_type_id IN (?)
             ORDER BY (mood_id IS NULL) ASC, mood_id ASC, female DESC,
                      unconverted DESC, glitched ASC, id DESC",
          Array [
            "2",
          ],
        ],
        Array [
          "SELECT sa.*, rel.parent_id FROM swf_assets sa
             INNER JOIN parents_swf_assets rel ON
               rel.parent_type = \\"PetState\\" AND
               rel.swf_asset_id = sa.id
             WHERE rel.parent_id IN (?)",
          Array [
            "17723",
          ],
        ],
        Array [
          "SELECT * FROM species_translations
             WHERE species_id IN (?) AND locale = \\"en\\"",
          Array [
            "54",
          ],
        ],
        Array [
          "SELECT * FROM color_translations
             WHERE color_id IN (?) AND locale = \\"en\\"",
          Array [
            "75",
          ],
        ],
        Array [
          "SELECT * FROM colors WHERE id IN (?) AND prank = 0",
          Array [
            "75",
          ],
        ],
        Array [
          "SELECT * FROM zones WHERE id IN (?,?,?,?,?,?)",
          Array [
            "15",
            "5",
            "37",
            "30",
            "33",
            "34",
          ],
        ],
      ]
    `);
  });

  it("loads multiple for species and color", async () => {
    const res = await query({
      query: gql`
        query {
          petAppearances(speciesId: "54", colorId: "75") {
            id

            species {
              id
              name
            }

            color {
              id
              name
            }

            bodyId
            petStateId
            pose
            layers {
              id
              imageUrl(size: SIZE_600)
              zone {
                depth
              }
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT * FROM pet_states
             WHERE pet_type_id IN (?)
             ORDER BY (mood_id IS NULL) ASC, mood_id ASC, female DESC,
                      unconverted DESC, glitched ASC, id DESC",
          Array [
            "2",
          ],
        ],
        Array [
          "SELECT sa.*, rel.parent_id FROM swf_assets sa
             INNER JOIN parents_swf_assets rel ON
               rel.parent_type = \\"PetState\\" AND
               rel.swf_asset_id = sa.id
             WHERE rel.parent_id IN (?,?,?,?,?,?,?,?)",
          Array [
            "17723",
            "17742",
            "5991",
            "436",
            "10014",
            "11089",
            "4751",
            "2",
          ],
        ],
        Array [
          "SELECT * FROM species_translations
             WHERE species_id IN (?) AND locale = \\"en\\"",
          Array [
            "54",
          ],
        ],
        Array [
          "SELECT * FROM color_translations
             WHERE color_id IN (?) AND locale = \\"en\\"",
          Array [
            "75",
          ],
        ],
        Array [
          "SELECT * FROM zones WHERE id IN (?,?,?,?,?,?)",
          Array [
            "15",
            "5",
            "37",
            "30",
            "33",
            "34",
          ],
        ],
      ]
    `);
  });

  it("loads unconverted appearance", async () => {
    const res = await query({
      query: gql`
        query {
          petAppearance(speciesId: "1", colorId: "63", pose: UNCONVERTED) {
            id

            species {
              id
              name
            }

            color {
              id
              name
              isStandard
            }

            layers {
              id
              imageUrl(size: SIZE_600)
              svgUrl
              zone {
                depth
              }
              restrictedZones {
                id
              }
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "1",
            "63",
          ],
        ],
        Array [
          "SELECT * FROM pet_states
             WHERE pet_type_id IN (?)
             ORDER BY (mood_id IS NULL) ASC, mood_id ASC, female DESC,
                      unconverted DESC, glitched ASC, id DESC",
          Array [
            "2274",
          ],
        ],
        Array [
          "SELECT sa.*, rel.parent_id FROM swf_assets sa
             INNER JOIN parents_swf_assets rel ON
               rel.parent_type = \\"PetState\\" AND
               rel.swf_asset_id = sa.id
             WHERE rel.parent_id IN (?)",
          Array [
            "2571",
          ],
        ],
        Array [
          "SELECT * FROM species_translations
             WHERE species_id IN (?) AND locale = \\"en\\"",
          Array [
            "1",
          ],
        ],
        Array [
          "SELECT * FROM color_translations
             WHERE color_id IN (?) AND locale = \\"en\\"",
          Array [
            "63",
          ],
        ],
        Array [
          "SELECT * FROM colors WHERE id IN (?) AND prank = 0",
          Array [
            "63",
          ],
        ],
        Array [
          "SELECT * FROM zones WHERE id IN (?)",
          Array [
            "46",
          ],
        ],
      ]
    `);
  });
});
