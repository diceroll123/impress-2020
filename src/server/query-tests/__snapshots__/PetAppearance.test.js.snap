// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetAppearance loads for species and color 1`] = `
Object {
  "petAppearance": Object {
    "color": Object {
      "id": "75",
      "isStandard": true,
      "name": "Starry",
    },
    "id": "17723",
    "layers": Array [
      Object {
        "id": "5995",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
        "restrictedZones": Array [],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/007/7941_2c4cc4b846/7941.svg",
        "zone": Object {
          "depth": 18,
        },
      },
      Object {
        "id": "5996",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
        "restrictedZones": Array [],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/007/7942_2eab06fd7b/7942.svg",
        "zone": Object {
          "depth": 7,
        },
      },
      Object {
        "id": "6000",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
        "restrictedZones": Array [],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/007/7946_0348dad587/7946.svg",
        "zone": Object {
          "depth": 40,
        },
      },
      Object {
        "id": "16467",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
        "restrictedZones": Array [],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/024/24008_a05fe9876a/24008.svg",
        "zone": Object {
          "depth": 34,
        },
      },
      Object {
        "id": "19784",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28892/600x600.png?v2-1313418652000",
        "restrictedZones": Array [],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/028/28892_a8e3a8b430/28892.svg",
        "zone": Object {
          "depth": 37,
        },
      },
      Object {
        "id": "178150",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/036/36887/600x600.png?v2-1354240708000",
        "restrictedZones": Array [],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/036/36887_a335fbba09/36887.svg",
        "zone": Object {
          "depth": 38,
        },
      },
    ],
    "species": Object {
      "id": "54",
      "name": "Zafara",
    },
  },
}
`;

exports[`PetAppearance loads multiple for species and color 1`] = `
Object {
  "petAppearances": Array [
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "17723",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
        Object {
          "id": "19784",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28892/600x600.png?v2-1313418652000",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "178150",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/036/36887/600x600.png?v2-1354240708000",
          "zone": Object {
            "depth": 38,
          },
        },
      ],
      "petStateId": "17723",
      "pose": "HAPPY_FEM",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "17742",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
        Object {
          "id": "19549",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28548/600x600.png?v2-1345719457000",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "178150",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/036/36887/600x600.png?v2-1354240708000",
          "zone": Object {
            "depth": 38,
          },
        },
      ],
      "petStateId": "17742",
      "pose": "HAPPY_MASC",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "5991",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "14790",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21057/600x600.png?v2-0",
          "zone": Object {
            "depth": 38,
          },
        },
        Object {
          "id": "14793",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21061/600x600.png?v2-0",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
      ],
      "petStateId": "5991",
      "pose": "SAD_FEM",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "436",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "14790",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21057/600x600.png?v2-0",
          "zone": Object {
            "depth": 38,
          },
        },
        Object {
          "id": "14792",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21060/600x600.png?v2-0",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
      ],
      "petStateId": "436",
      "pose": "SAD_MASC",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "10014",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "14791",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21059/600x600.png?v2-0",
          "zone": Object {
            "depth": 38,
          },
        },
        Object {
          "id": "14795",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21066/600x600.png?v2-0",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
      ],
      "petStateId": "10014",
      "pose": "SICK_FEM",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "11089",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "14791",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21059/600x600.png?v2-0",
          "zone": Object {
            "depth": 38,
          },
        },
        Object {
          "id": "14794",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/021/21064/600x600.png?v2-0",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
      ],
      "petStateId": "11089",
      "pose": "SICK_MASC",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "4751",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
        Object {
          "id": "19550",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28549/600x600.png?v2-0",
          "zone": Object {
            "depth": 38,
          },
        },
        Object {
          "id": "19784",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28892/600x600.png?v2-1313418652000",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "163528",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28549/600x600.png?v2-1326455337000",
          "zone": Object {
            "depth": 38,
          },
        },
      ],
      "petStateId": "4751",
      "pose": "UNKNOWN",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
    Object {
      "bodyId": "180",
      "color": Object {
        "id": "75",
        "name": "Starry",
      },
      "id": "2",
      "layers": Array [
        Object {
          "id": "5995",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7941/600x600.png?v2-0",
          "zone": Object {
            "depth": 18,
          },
        },
        Object {
          "id": "5996",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7942/600x600.png?v2-0",
          "zone": Object {
            "depth": 7,
          },
        },
        Object {
          "id": "6000",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/007/7946/600x600.png?v2-0",
          "zone": Object {
            "depth": 40,
          },
        },
        Object {
          "id": "16467",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/024/24008/600x600.png?v2-0",
          "zone": Object {
            "depth": 34,
          },
        },
        Object {
          "id": "19549",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28548/600x600.png?v2-1345719457000",
          "zone": Object {
            "depth": 37,
          },
        },
        Object {
          "id": "19550",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28549/600x600.png?v2-0",
          "zone": Object {
            "depth": 38,
          },
        },
        Object {
          "id": "163528",
          "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/028/28549/600x600.png?v2-1326455337000",
          "zone": Object {
            "depth": 38,
          },
        },
      ],
      "petStateId": "2",
      "pose": "UNKNOWN",
      "species": Object {
        "id": "54",
        "name": "Zafara",
      },
    },
  ],
}
`;

exports[`PetAppearance loads unconverted appearance 1`] = `
Object {
  "petAppearance": Object {
    "color": Object {
      "id": "63",
      "isStandard": true,
      "name": "Royalboy",
    },
    "id": "2571",
    "layers": Array [
      Object {
        "id": "9941",
        "imageUrl": "https://impress-asset-images.s3.amazonaws.com/biology/000/000/013/13024/600x600.png?v2-0",
        "restrictedZones": Array [
          Object {
            "id": "4",
          },
          Object {
            "id": "5",
          },
          Object {
            "id": "6",
          },
          Object {
            "id": "7",
          },
          Object {
            "id": "8",
          },
          Object {
            "id": "9",
          },
          Object {
            "id": "10",
          },
          Object {
            "id": "11",
          },
          Object {
            "id": "12",
          },
          Object {
            "id": "13",
          },
          Object {
            "id": "14",
          },
          Object {
            "id": "15",
          },
          Object {
            "id": "16",
          },
          Object {
            "id": "17",
          },
          Object {
            "id": "18",
          },
          Object {
            "id": "19",
          },
          Object {
            "id": "20",
          },
          Object {
            "id": "21",
          },
          Object {
            "id": "22",
          },
          Object {
            "id": "23",
          },
          Object {
            "id": "24",
          },
          Object {
            "id": "25",
          },
          Object {
            "id": "26",
          },
          Object {
            "id": "27",
          },
          Object {
            "id": "28",
          },
          Object {
            "id": "29",
          },
          Object {
            "id": "30",
          },
          Object {
            "id": "31",
          },
          Object {
            "id": "32",
          },
          Object {
            "id": "33",
          },
          Object {
            "id": "34",
          },
          Object {
            "id": "35",
          },
          Object {
            "id": "36",
          },
          Object {
            "id": "37",
          },
          Object {
            "id": "38",
          },
          Object {
            "id": "39",
          },
          Object {
            "id": "40",
          },
          Object {
            "id": "41",
          },
          Object {
            "id": "42",
          },
          Object {
            "id": "43",
          },
        ],
        "svgUrl": "http://images.neopets.com/cp/bio/data/000/000/013/13024_18911a85d1/13024.svg",
        "zone": Object {
          "depth": 48,
        },
      },
    ],
    "species": Object {
      "id": "1",
      "name": "Acara",
    },
  },
}
`;
